packages/api-contract/src/types.ts

```diff
@@ -73,7 +73,7 @@ export interface DecodedMessage {
 }
 
 export interface ContractOptions {
-  gasLimit?: bigint | string | number | BN | WeightV2;
+  gasLimit?: bigint | string | number | BN | Weight;
```

Comment:
This one should indeed be `WeightV2`. There is internal detection from the options to cater for both old/new (aka if a `number` is passed, it would convert to `WeightV2` before using.

Reply:
Sorry I misunderstand your means, I've reverted this change